tlatools/org.lamport.tlatools/src/tla2sany/drivers/SANY.java

```diff
@@ -193,73 +186,55 @@ public static void frontEndInitialize(SpecObj spec, PrintStream syserr )
     catch (Exception e) {
       syserr.println("Unexpected exception during SANY initialization " +
                      fileName + "\n" + e);
-      // syserr.println("\nStack trace for exception:\n"); 
-      // e.printStackTrace(syserr);
       syserr.println("Initialization errors detected before " + 
                      "the unexpected exception:\n");
-      syserr.print( initErrors );
+      syserr.print(initErrors);
 
       // indicate fatal error during first phase
       spec.errorLevel = 1;  
       throw new InitException();
     }
-    return;
-  } // frontEndInitialize
+  }
 
   // Parse all of the files referred to by the top-level file in specification
   public static void frontEndParse(SpecObj spec, PrintStream syserr) 
   throws ParseException {
 	  frontEndParse(spec, syserr, true);
   }
-  public static void frontEndParse(SpecObj spec, PrintStream syserr, boolean validatePCalTranslation) 
+  public static void frontEndParse(SpecObj spec, PrintStream syserr, boolean validatePCalTranslation)
   throws ParseException {
-      /***********************************************************************
-       * Modified on 12 May 2008 by LL to remove "throws AbortException",     *
-       * since it catches all exceptions and turns them into                  *
-       * ParseExceptions.                                                     *
-       ***********************************************************************/
-      try 
+      Objects.requireNonNull(spec);
+      Objects.requireNonNull(syserr);
```

Comment:
Validated the input once to remove the if checks around syserr (which in some places were also missing).